src/Compiler/Lambda/OptLambda.sml

```diff
@@ -246,13 +246,20 @@ structure OptLambda: OPT_LAMBDA =
 		     Type=ARROWtype(argtypes,[restype])}
       fun f256_bin opr (x:exp,y:exp) : exp =
           PRIM(ccall ("__" ^ opr ^ "_f256") [f256Type,f256Type] f256Type, [x,y])
+      val allocVector : exp =
+        let val zero = PRIM(ccall "__real_to_f64" [realType] f64Type, [REAL ("0.0", NONE)])
+             in PRIM(BLOCKF64prim, [zero, zero, zero, zero])
+        end
     in
-      fun f256_box (x:exp) : exp = PRIM(ccall "__f256_box" [f256Type] stringType, [x])
+      fun f256_box (x:exp) : exp = PRIM(ccall "__f256_box" [f256Type] stringType,[x])
```

Comment:
I don't like the overloading of the primitive `__f256_box` - delete the function `fun f256_box ...`.